pkg/processor/metricfrequencyprocessor/metriccache.go

```diff
@@ -0,0 +1,84 @@
+package metricfrequencyprocessor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+const (
+	dataPointExpiration           = time.Hour * 1
+	dataPointCacheCleanupInterval = time.Minute * 10
+	metricCacheCleanupInterval    = time.Hour * 3
+)
+
+type DataPoint struct {
+	Timestamp pdata.Timestamp
+	Value     float64
+}
+
+// metricCache caches data points in two level mapping structure.
+// To easily list all data points of a given metric it keeps a separate cache for each incoming metric.
+type metricCache struct {
+	internalCaches map[string]*cache.Cache
+}
+
+func newMetricCache() *metricCache {
+	c := &metricCache{
+		internalCaches: make(map[string]*cache.Cache),
+	}
+
+	go func(c *metricCache) {
+		t := time.NewTicker(metricCacheCleanupInterval)
+		defer t.Stop()
+		for range t.C {
+			c.Cleanup()
+		}
+	}(c)
+
+	return c
+}
+
+func (mc *metricCache) Register(name string, dataPoint pdata.DoubleDataPoint) {
+
+	internalCache, exists := mc.internalCaches[name]
+	if !exists {
```

Comment:
nit, normally `ok` is used:

```suggestion
	internalCache, ok := mc.internalCaches[name]
	if !ok {
```